### DISCUSSION TITLE
tt-weave and "XeTeX: A Pseudoprogram"

### DISCUSSION THREAD

**Discussion by shreevatsa** [shreevatsa/webWEB#23] [General]
(TLDR: See [tt-weave](https://github.com/tectonic-typesetting/tt-weave) and [XeTeX: A Pseudoprogram](https://stacks.fullyjustified.net/xap/2022.0/).)

[Peter Williams](https://newton.cx/~peter/) (@pkgw), gave talks about his [Tectonic](https://tectonic-typesetting.github.io/en-US/) project at [TUG 2022](https://tug.org/tug2022/) in July ([video](https://www.youtube.com/watch?v=n2htt360liY)), and at [TeX Hour 2022-09-29](https://texhour.github.io/2022/09/29/rethink-tex-in-stem/), in which, in passing, he mentioned a project to parse xetex.web and render it in HTML. This project has now (in October) been made public as [tt-weave](https://github.com/tectonic-typesetting/tt-weave), with the result as [XeTeX: A Pseudoprogram](https://stacks.fullyjustified.net/xap/2022.0/) (from `xetex.web`) and [The WEAVE Processor](https://stacks.fullyjustified.net/weave/2022.0/) (from `weave.web`) ([announcement](https://github.com/tectonic-typesetting/tectonic/discussions/952)).

I haven't had the chance to look closely yet but from first impressions it looks fantastic, and pretty much exactly the topic of this discussion repo. If anyone is reading this and is interested, please take a look! :-)

**Comment by pkgw** [comment]
Thanks for the mention! Happy to answer any questions about it. There is definitely room for improvement in the web app frontend — a lot of the UI features are prototype-quality. (I've been spending my time on a next-step project that will hopefully feed back to the next iteration of tt-weave, but isn't ready yet.)
